darts/utils/likelihood_models.py

```diff
@@ -0,0 +1,71 @@
+"""
+Likelihood Models
+-----------------
+"""
+
+from abc import ABC, abstractmethod
+from typing import Sequence, Optional, Tuple
+import torch
+import torch.nn as nn
+
+from ..timeseries import TimeSeries
+
+
+class LikelihoodModel(ABC):
+
+    def __init__(self):
+        """
+        Abstract class for a likelihood model. It contains all the logic to compute the loss
+        and to sample the distribution, given the parameters of the distribution
+        """
+        pass
+
+    @abstractmethod
+    def _compute_loss(self, output: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+        """
+        Computes a loss from a model `output`, which represents the parameters of a given probability
+        distribution for every ground truth value in `target`, and the `target` itself.
+        """
+        pass
+
+    @abstractmethod
+    def _sample(self, output: torch.Tensor) -> torch.Tensor:
+        """
+        Samples a prediction from the probability distributions defined by the specific likelihood model
+        and the parameters given in `output`.
```

Comment:
It's slightly weird to name it `output`, because even though most of the time it's the output of a network, from the point of view of this module what's received here is more of an input. But this is really a detail, no need to change (unless you really want to).

Reply:
True. How about `model_output` or `likelihood_parameters`?

Reply:
model_output is good 👍